jobrunner/config.py

```diff
@@ -100,3 +102,35 @@ class ConfigException(Exception):
     "STATA_LICENSE_REPO",
     "https://github.com/opensafely/server-instructions.git",
 )
+
+
+def parse_job_resource_weights(config_file):
+    """
+    Parse a simple ini file which looks like this:
+
+        [some-workspace-name]
+        my-ram-hungry-action = 4
+        other-actions.* = 1.5
+
+        [other-workspace-name]
+        ...
+
+    Any jobs in the specified workspace will have their action names matched
+    against the regex patterns specified in the config file and will be
+    assigned the weight of the first matching pattern. All other jobs are
+    assigned a weight of 1.
+    """
+    config_file = Path(config_file)
+    weights = {}
+    if config_file.exists():
+        config = configparser.ConfigParser()
+        config.read_string(config_file.read_text(), source=str(config_file))
+        for workspace in config.sections():
+            weights[workspace] = {
+                re.compile(pattern): float(weight)
+                for (pattern, weight) in config.items(workspace)
+            }
+    return weights
+
+
+JOB_RESOURCE_WEIGHTS = parse_job_resource_weights("job-resource-weights.ini")
```

Comment:
Local config file is fine for now, and you've probably already considered this, but:

I think in future the weight should come in from the job-server with the JobRequest.  Doing so would work across backends, and allow us to automatically generate those weights based on stats in the future without changing the job-runner. We could also allow users to prioritse amongst their own (or their groups) queued jobs in the same way.

Reply:
Yes, I think putting the job-server in control of this makes a lot of sense longer term, assuming it has access to the necessary data. The weights thing was just a quick and dirty way of solving the immediate problem of not running too many high memory jobs at the same time and I'm not sure we'll keep using that. I'd imagine we'll track expected RAM, CPU and disk usage independently and make decisions based on availability of specific resources.